File: examples-sdk/go/017-basic-production-fullstack/main.go
// 017-basic-production-fullstack — Go SDK equivalent
// Compile: vil compile --from go --input 017-basic-production-fullstack/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("production-fullstack", 8080)

	fullstack := vil.NewService("fullstack")
	fullstack.Endpoint("GET", "/stack", "stack_info")
	fullstack.Endpoint("GET", "/config", "full_config")
	fullstack.Endpoint("GET", "/sprints", "sprints")
	fullstack.Endpoint("GET", "/middleware", "middleware_info")
	s.Service(fullstack)

	admin := vil.NewService("admin")
	admin.Endpoint("GET", "/config", "full_config")
	s.Service(admin)

	root := vil.NewService("root")
	s.Service(root)

	s.Compile()
}

File: examples-sdk/zig/017-basic-production-fullstack/main.zig
// 017-basic-production-fullstack — Zig SDK equivalent
// Compile: vil compile --from zig --input 017-basic-production-fullstack/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("production-fullstack", 8080);
    var fullstack = vil.Service.init("fullstack");
    fullstack.endpoint("GET", "/stack", "stack_info");
    fullstack.endpoint("GET", "/config", "full_config");
    fullstack.endpoint("GET", "/sprints", "sprints");
    fullstack.endpoint("GET", "/middleware", "middleware_info");
    server.service(&fullstack);
    var admin = vil.Service.init("admin");
    admin.endpoint("GET", "/config", "full_config");
    server.service(&admin);
    var root = vil.Service.init("root");
    server.service(&root);
    server.compile();
}
